src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -139,7 +143,7 @@ private HttpRequestMessage CreateRequestMessage(Uri endpoint, IDictionary<string
                     (int)response.StatusCode, response.StatusCode));
 
                 is5xxError = (int)response.StatusCode >= 500 && (int)response.StatusCode < 600;
-                isRetryable = is5xxError && !HasRetryAfterHeader(response);
+                isRetryable = is5xxError && !HasRetryAfterHeader(response) && _retryConfig;
```

Comment:
Nit:
might be more optimized to go from the lease expensive bool checks (_retryConfig) to the most expensive checks (HasRetryAfterHeader).

But don't have strong opinions as the effect is fairly negligible.